wellcomeml/ml/vectorizer.py

```diff
@@ -30,17 +26,26 @@ def __init__(self, embedding="tf-idf", cache_transformed=False, **kwargs):
         self.embedding = embedding
         self.cache_transformed = cache_transformed
 
+        # Only actually import when necessary (extras might not be installed)
+
         vectorizer_dispatcher = {
-            "tf-idf": WellcomeTfidf,
-            "bert": BertVectorizer,
-            "keras": KerasVectorizer,
-            "doc2vec": Doc2VecVectorizer,
+            "tf-idf": "wellcomeml.ml.frequency_vectorizer.WellcomeTfidf",
+            "bert": "wellcomeml.ml.bert_vectorizer.BertVectorizer",
+            "keras": "welclomeml.ml.keras_vectorizer.KerasVectorizer",
+            "doc2vec": "wellcomeml.ml.doc2vec_vectorizer.Doc2VecVectorizer",
         }
 
         if not vectorizer_dispatcher.get(embedding):
             raise ValueError(f"Model {embedding} not available")
 
-        self.vectorizer = vectorizer_dispatcher.get(embedding)(**kwargs)
+        vectorizer_path = '.'.join(vectorizer_dispatcher.get(embedding).split('.')[:-1])
+        vectorizer_class_path = vectorizer_dispatcher.get(embedding).split('.')[-1]
+        vectorizer = getattr(
+            __import__(vectorizer_path, fromlist=[vectorizer_class_path]),
```

Comment:
should't you try to import first and throw an error in case user does not have the library needed?

Reply:
Yes, that's what will happen because those modules have now the `throw_extra_import_messages`.

What I wanted to avoid, in addition, was to import all the modules unnecessarily. Say I have everything installed but just want to use tf-idf - then it just imports it.

Reply:
👍